Result of `nixpkgs-review pr 107653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python37Packages.gpy</li>
    <li>python37Packages.gpyopt</li>
  </ul>
</details>
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>python37Packages.paramz</li>
    <li>python38Packages.gpy</li>
    <li>python38Packages.gpyopt</li>
    <li>python38Packages.paramz</li>
    <li>python39Packages.gpy</li>
    <li>python39Packages.gpyopt</li>
    <li>python39Packages.paramz</li>
  </ul>
</details>

python37Packages.gpy
```
running install tests
...................................F........... /nix/store/c7y36xrsgm31phhi786sshksj47pwll6-python3.7-paramz-0.9.5/lib/python3.7/site-packages/paramz/transformations.py:111: RuntimeWarnin>... /build/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 1 then the given X dimension 2. Be very sure this is what you want and you have not forgott>............... /build/GPy-1.9.9/GPy/core/gp.py:87: UserWarning:Your kernel has a different input dimension 4 then the given X dimension 5. Be very sure this is what you want and you have>............... /build/GPy-1.9.9/GPy/util/warping_functions.py:142: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:143: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:144: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:152: RuntimeWarning:overflow encountered in cosh
 /build/GPy-1.9.9/GPy/util/warping_functions.py:153: RuntimeWarning:overflow encountered in cosh
...
======================================================================
FAIL: Testing the TP regression with matern32 kernel on 1d data
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/GPy-1.9.9/GPy/testing/model_tests.py", line 843, in test_TPRegression_matern32_1D
    self.check_model(matern32, model_type='TPRegression', dimension=1)
  File "/build/GPy-1.9.9/GPy/testing/model_tests.py", line 682, in check_model
    self.assertTrue(m.checkgrad())
AssertionError: False is not true

----------------------------------------------------------------------
Ran 83 tests in 239.213s

FAILED (failures=1)
```

--- a/pkgs/development/python-modules/gpy/default.nix
+++ b/pkgs/development/python-modules/gpy/default.nix
@@ -1,5 +1,14 @@
-{ stdenv, buildPythonPackage, fetchPypi
-, numpy, scipy, six, paramz, nose, matplotlib, cython }:
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, numpy
+, scipy
+, six
+, paramz
+, matplotlib
+, cython
+, nose
+}:
 
 buildPythonPackage rec {
   pname = "GPy";
@@ -10,21 +19,35 @@ buildPythonPackage rec {
     sha256 = "04faf0c24eacc4dea60727c50a48a07ddf9b5751a3b73c382105e2a31657c7ed";
   };
 
-  # running tests produces "ImportError: cannot import name 'linalg_cython'"
-  # even though Cython has run
-  checkPhase = "nosetests -d";
-  doCheck = false;
-
+  buildInputs = [ cython ];
+  propagatedBuildInputs = [ numpy scipy six paramz matplotlib ];
   checkInputs = [ nose ];
 
-  buildInputs = [ cython ];
+  # $ nosetests GPy/testing/*.py
+  # => Ran 483 tests in 112.146s (on 8 cores)
+  # So instead, run shorter set of tests
+  checkPhase = ''
+    nosetests GPy/testing/linalg_test.py
+  '';
 
-  propagatedBuildInputs = [ numpy scipy six paramz matplotlib ];
+  # Rebuild cython-generated .c files since the included
+  # ones were built with an older version of cython that is
+  # incompatible with python3.9
+  preBuild = ''
+    for fn in $(find . -name '*.pyx'); do
+      echo $fn | sed 's/\.\.pyx$/\.c/' | xargs ${cython}/bin/cython -3
+    done
+  '';
+
+  pythonImportsCheck = [
+    "GPy"
+  ];
 
   meta = with stdenv.lib; {
     description = "Gaussian process framework in Python";
     homepage = "https://sheffieldml.github.io/GPy";
     license = licenses.bsd3;
     maintainers = with maintainers; [ bcdarwin ];
+    broken = stdenv.isDarwin;  # See inscrutable error message here: https://github.com/NixOS/nixpkgs/pull/107653#issuecomment-751527547
   };
 }

--- a/pkgs/development/python-modules/paramz/default.nix
+++ b/pkgs/development/python-modules/paramz/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, buildPythonPackage, fetchPypi, numpy, scipy, six, decorator }:
+{ stdenv, buildPythonPackage, fetchPypi, numpy, scipy, six, decorator, nose }:
 
 buildPythonPackage rec {
   pname = "paramz";
@@ -10,6 +10,12 @@ buildPythonPackage rec {
   };
 
   propagatedBuildInputs = [ numpy scipy six decorator ];
+  checkInputs = [ nose ];
+
+  # Ran 113 tests in 3.082s
+  checkPhase = ''
+      nosetests -v paramz/tests
+  '';
 
   meta = with stdenv.lib; {
     description = "Parameterization framework for parameterized model creation and handling";

